### PR TITLE
#150 support event over three days.

### PR DESCRIPTION
**Pull request by Vanco**
Add unique check based on the event id before add to cache, and and support the event which cross over more than three days.

### CODE REVIEW

File: library/src/main/java/com/alamkanak/weekview/WeekViewEvent.java [hunk 1]
```diff
@@ -106,4 +106,20 @@ public long getId() {
...
+        WeekViewEvent that = (WeekViewEvent) o;
+
+        return mId == that.mId;
+
+    }
+
+    @Override
+    public int hashCode() {
+        return (int) (mId ^ (mId >>> 32));
+    }
```
Comment (alamkanak) [inline library/src/main/java/com/alamkanak/weekview/WeekViewEvent.java]: May I ask where are we using `equals` and `hashCode` method?

Comment (lolobosse) [inline library/src/main/java/com/alamkanak/weekview/WeekViewEvent.java]: He said he wants to check for the unicity of every appmt so I guess it is for that reason. Kind of the same idea as mine #109 (but he added the long appmt feature which is cool)

Comment (Vanco) [inline library/src/main/java/com/alamkanak/weekview/WeekViewEvent.java]: I actually implicit use the `equals` and `hashCode`  in the `MonthChangeListener` when we feed data in.  we need async call our server to get event data, and feed in. Unfortunately, the onMonthChange are called sync way.  So we make a pain implement to make it happen. We Use Set to make sure same event only add once into the Set.  So the WeekView can display correctly.  That's not good, but just work for me.

```
// cache set of event
private Set<WeekViewEvent> events=null;

// async call our server to get event data
WeekView.MonthChangeListener mMonthChangeListener = new WeekView.MonthChangeListener() {
        public List<WeekViewEvent> onMonthChange(int newYear, int newMonth) {
        // ... call server asnyc
        
RequstServer.getInstance(this).getEvent(request, new Succeed<Response>() {
            @Override
            public void onSucced(Response object) {
                
                    for(int i=0;i<object.getList().size();i++){
                        WeekViewEvent event=new WeekViewEvent();
                        
                        // ...... populate event and add to Set.
                        
                        events.add(event);
                    }
                    weekView.notifyDatasetChanged();
                }
            }
        });

        ArrayList<WeekViewEvent> evs = new ArrayList<>();
            evs.addAll(events);
            return evs;
     }
    };
```

**Comment by caske33** [comment]
This is now merged into `develop` as `853ca78eedc25dcdfe52fa5685f5c16157d5b50c`

**Pull request closed without merging**